### PR TITLE
chore(http/github): add utility function to fetch raw files

### PR DESCRIPTION
**Pull request by secustor**
## Changes
add `getRawFile` which will set the necessary accept header to get a raw file. 
If the requested API path is not a single file it will default to the same behavior as `application/json`

## Context
Enables https://github.com/renovatebot/renovate/pull/30155

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/github.spec.ts [hunk 1]
```diff
@@ -786,4 +786,24 @@ describe('util/http/github', () => {
...
+        .scope(githubApiHost)
+        .get('/foo/bar/contents/lore/ipsum.txt')
+        .matchHeader(
+          'accept',
+          'application/vnd.github.raw+json, application/vnd.github.v3+json',
+        )
+        .reply(200, 'foo');
+      await expect(
+        githubApi.getRawFile(
+          `${githubApiHost}/foo/bar/contents/lore/ipsum.txt`,
```
Comment (viceice) [inline lib/util/http/github.spec.ts:802]: will it work with relative paths?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/util/http/github.spec.ts (hunk 1 above)
Comment (secustor) [reply lib/util/http/github.spec.ts:802]: Yes, it does.

File: lib/util/http/github.spec.ts [hunk 2]
```diff
@@ -786,4 +786,38 @@ describe('util/http/github', () => {
...
+    });
+
+    it('support relative path', async () => {
+      httpMock
+        .scope(githubApiHost)
+        .get('/foo/bar/contents/lore/ipsum.txt')
+        .reply(200, 'foo');
+      await expect(
+        githubApi.getRawFile(
+          `${githubApiHost}/foo/bar/contents/foo/../lore/ipsum.txt`,
```
Comment (viceice) [inline lib/util/http/github.spec.ts:816]:
```suggestion
          `foo/bar/contents/foo/../lore/ipsum.txt`,
```
I mean this

Comment (secustor) [reply lib/util/http/github.spec.ts:816]: No that isn't supported

Comment (secustor) [reply lib/util/http/github.spec.ts:820]: Actually, it has worked already. 

I have added additionally logic to allow simply querying by providing the path to a file and the repository as option.

File: lib/util/http/github.ts [hunk 3]
```diff
@@ -493,4 +493,24 @@ export class GithubHttp extends Http<GithubHttpOptions> {
...
+      headers: {
+        accept: 'application/vnd.github.raw+json',
+      },
+    };
+
+    let newURL = url;
+    const httpRegex = regEx(/^https?:\/\//);
+    if (options.repository && !httpRegex.test(options.repository)) {
+      newURL = joinUrlParts(options.repository, 'contents', url);
+    }
```
Comment (viceice) [inline lib/util/http/github.ts:512]: why do we need this? only baseUrl should be added if url isn't absolute. 🤔

the repo field is used for auth only, see request method

Comment (secustor) [reply lib/util/http/github.ts:512]: Or do you simply mean `repository` should be removed from `newOptions` so it is not fed into `request`?

Comment (viceice) [reply lib/util/http/github.ts:512]: No, my idea with relative paths where to allow `foo/bar/contents/lore/ipsum.txt` urls, which should just work

File: lib/util/http/github.ts [hunk 4]
```diff
@@ -493,4 +493,24 @@ export class GithubHttp extends Http<GithubHttpOptions> {
 
     return result;
   }
+
+  public getRawFile(
```
Comment (viceice) [inline lib/util/http/github.ts:497]: Add a tsdoc comment and a note that only text files are allowed. The return type indicates a string body, so when you pass a binary file url, then you'll get a Buffer as body which is higly unexpected at runtime.

So maybe rename to explpict mention and validate in function that returned body is a string.
```suggestion
  public getRawTextFile(
```

File: lib/util/http/github.ts (hunk 3 above)
Comment (secustor) [reply lib/util/http/github.ts:523]: Yeah, they work already. I have added some tests to prevent regressions.

File: lib/util/http/github.ts (hunk 4 above)
Comment (secustor) [reply lib/util/http/github.ts]: I haven't managed to force got to return a Buffer without setting the `responseType`, though I have added a check and a test for it just to be sure.

**Review by viceice — APPROVED** [review]

**Pull request merged**